ksql-engine/src/main/java/io/confluent/ksql/function/Blacklist.java

```diff
@@ -26,31 +26,30 @@
 import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
 import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
 /**
  * Used to restrict the classes that can be loaded by user supplied UDFs
  */
 public class Blacklist implements Predicate<String> {
   private static final Logger logger = LoggerFactory.getLogger(Blacklist.class);
-  private static final String EMPTY_BLACKLIST = "^(?)\\.?.*$";
+  private static final String EMPTY_BLACKLIST = "^(?:)\\.?.*$";
```

Comment:
Split into two constants: 

```
private static final String BLACKLIST_PREFIX = "^(?:";
private static final String BLACKLIST_SUFFIX = ")\\.?.*$";
```

Then your collect line can be:

```
.collect(Collectors.joining("|",  BLACKLIST_PREFIX, BLACKLIST_SUFFIX));
```

And your empty check can be:

```
if (this.blackList.equals(BLACKLIST_PREFIX + BLACKLIST_SUFFIX))
```

Which I think is easier to understand.